### ISSUE TITLE
Access denied for user 'froxlor'@'localhost'

### ISSUE THREAD

**Issue by torvaldhessel** [Froxlor/Froxlor#963]
**Describe the bug**
Installed Froxlor successfully on a VPS with Ubuntu 20.04, and can browse to the Froxlor web-install on local host. 
Filled in all fields, and upon submit:
SQLSTATE[HY000] [1044] Access denied for user 'froxlor'@'127.0.0.1' to database 'froxlor'

**System information**
* Froxlor version: latest (downloaded yesterday 07/24)
* Web server: Apache/2.4.41 (Ubuntu)
* OS/Version: Ubuntu 20.04

**To Reproduce**
-install as instructed
-browse to froxlor on the server's IP address
-follow instructions & press submit

Result:
Checking MySQL-root access... | OK
Creating backup of old database... | OK (/tmp/froxlor_backup_202107241336.sql)
Preparing database... | OK
Creating database and username... | OK
Testing if database and user have been created correctly... | SQLSTATE[HY000] [1044] Access denied for user 'froxlor'@'127.0.0.1' to database 'froxlor'

**Expected behavior**
Successful install :-/

**Logfiles**
Checking in Mysql, the froxlor users (three of them) did get created (one on localhost, one on 127.0.0.1 and one on the server's IP address). I have given all three ALLL PRIVILEGES as well but no change in result

I am relatively new to Linux/Ubuntu so if there is a log file somewhere, please give me the steps to get the file.

**Comment by d00p** [comment]
What DBMS exactly? MySQL? Mariadb? Which Version?

**Comment by torvaldhessel** [comment]
Running MySQL:
Your MySQL connection id is 13
Server version: 8.0.25-0ubuntu0.20.04.1 (Ubuntu)

**Comment by d00p** [comment]
The tests are all good for mysql8, trying to reproduce that on a testing machine, might take a moment

**Comment by d00p** [comment]
I cannot reproduce this. Installed a fresh ubuntu 20.04 with mysql8 - left everything as the default suggested. Went trough the installation and it worked just fine. No issues at all.

**Comment by torvaldhessel** [comment]
Michael, thank you for checking. I must have installed something that is causing this. I am rebuilding the server as we speak with a fresh copy, and start with just the minimum reqs for Froxlor. Thanks again for your effort.

**Comment by d00p** [comment]
If the problem persists for you, we might be able to help in the Forum or IRC

**Comment by torvaldhessel** [comment]
OK wiped the server, reinstalled Ubuntu 20.04 (Godaddy VPS)
-Did a minimal LAMP install (php 7.4.3)
-Reinstalled Foxlor
-Same issue

-Disabled firewall (on a hunch): Same issue

![foxlor](https://user-images.githubusercontent.com/79284574/126907720-f3e29f2f-07ba-4a98-a64f-ab3e8039d808.png)

-Wiped server and reinstall again
-Install with MariaDB instead of MySQL
-After several tries it turns out I cannot install MariaDB on a Godaddy VPS. After installing and then rebooting, mariaDB cannot be found, PID files do not exist etc.

Soooo..... after yet another wiping of the server, I am back at my original error. :-(

If IRS or forum help is available, I will certainly take it. My only other option at this point is to find an alternative to Foxlor, but looking at them they seem lesser options

**Comment by d00p** [comment]
Did you try localhost instead of 127.0.0.1?
How did you install mysql8? Official Ubuntu sources or via MySQL repository? Never had any issues with mariadb on Ubuntu. Maybe some special System-Setup by godaddy?

**Comment by torvaldhessel** [comment]
yes I tried both local host as well as 127.0.0.1 (even tried the server's IP address ;-) )

As for the MySql install, I followed these steps: https://www.digitalocean.com/community/tutorials/how-to-install-mysql-on-ubuntu-20-04

And as for MariaDb... Yes I think you're spot on and Godaddy has some other added dependencies etc. Right now I think that the best guess is that our bug here is also due to something Godaddy installs/blocks/whatever

**Comment by d00p** [comment]
Did you also manually create the froxlor user for the database or just root?

**Comment by torvaldhessel** [comment]
I created a new user with all privileges and used that instead of the pre-filled 'root' account. The froxlor user seemed to get added automatically upon install in localhost, 127.0.01 and the server IP address (so three users truly) and afterwards I see the users in the Mysql list. I also tried giving all three root privileges so I know they can do anything they need to, but no change. :-(

**Comment by d00p** [comment]
Yes, the unprivileged user will be created. Can you try using the real root user for the root account to know whether it has something to do with it

**Comment by torvaldhessel** [comment]
hmmmm maybe we're getting somewhere. I cannot login as root. I set up a secondary account (intended for Froxlor) with root privileges, so I can get in, but root cannot. Logged in with the alt. root user, changed root's password, and I still cannot get in.

OK let me go down that route for a bit and see if I can figure out why the root account behaves that way.

Just FYI I am startiung a new job today, so I will not have too much time to work on this. But I will continue and let you know. Very much appreciate your help and suggestions.

**Comment by torvaldhessel** [comment]
OK Success!

In my new job (quite unexpected) everyone works on Linux instead of Windows, but they use Debian. So, I decided to rebuild the VPS with Debian as well and see if that makes any difference => same error.

But then on this forum: https://forum.froxlor.org/index.php?/topic/14103-sqlstatehy000-1045-access-denied-for-user/ I saw the advice to check if the dbfoxlor user has access by using "# mysql -u dbfroxlor -p -h localhost" in my case the answer was "no" (note my error did not refer to dbfroxlor but simply froxlor@localhost, so I manually added both users (dbfroxlor and froxlor) and gave them root privileges (for now)

And that worked.
